core-spi-test/src/main/java/org/ehcache/internal/store/StorePutIfAbsentTest.java

```diff
@@ -143,9 +144,8 @@ public void wrongKeyTypeThrowsException()
         kvStore2.putIfAbsent("key", value);
       }
       throw new AssertionError("Expected ClassCastException because the key is of the wrong type");
-    } catch (ClassCastException e) {
-      // expected
-    } catch (CacheAccessException e) {
+    } catch (CacheAccessException cae) {
+      assertThat(cae.getCause(), instanceOf(ClassCastException.class));
```

Comment:
Wait, that changes the test semantics. You should do nothing if the cause is a `ClassCastException` and throw otherwise.